src/Traversals/Traversals.jl

```diff
@@ -15,122 +28,164 @@ depth-first traversal (`DepthFirst`) for the various traversal functions.
 per edge, depending on the traversal algorithm, for a function that operates on the first occurrence
-only, use [`newvisitfn!`](@ref).
+only, use [`newvisitfn!`](@ref). For parallel algorithms, the thread ID or the CPU ID will be passed
+in `t`.
 """
-@inline visitfn!(::AbstractTraversalState, u, v) = true
+@inline visitfn!(::TraversalState, u, v) = true
+@inline visitfn!(s::TraversalState, u, v, t::Integer) = visitfn!(s, u, v)
 
 """
     postvisitfn!(state, u)
+    postvisitfn!(state, u, t)
 
-Modify [`AbstractTraversalState`](@ref) `state` after having examined all neighbors of vertex `u`,
-and return `true` if successful; `false` otherwise.
+Modify [`TraversalState`](@ref) `state` after having examined all neighbors of vertex `u`,
+and return `true` if successful; `false` otherwise. For parallel algorithms, the thread ID or the
+CPU ID will be passed in `t`.
 """
-@inline postvisitfn!(::AbstractTraversalState, u) = true
+@inline postvisitfn!(::TraversalState, u) = true
+@inline postvisitfn!(s::TraversalState, u, t::Integer) = postvisitfn!(s, u)
+
 """
     postlevelfn!(state)
 
-Modify [`AbstractTraversalState`](@ref) `state` before moving to the next vertex in the traversal algorithm,
+Modify [`TraversalState`](@ref) `state` before moving to the next vertex in the traversal algorithm,
 and return `true` if successful; `false` otherwise.
 """
-@inline postlevelfn!(::AbstractTraversalState) = true
+@inline postlevelfn!(::TraversalState) = true
 
 ##############
 # functions common to both BreadthFirst and DepthFirst
 ##############
 """
-    traverse_graph!(g, s, alg, state)
     traverse_graph!(g, ss, alg, state)
 
-Traverse a graph `g` from source vertex `s` / vertices `ss` keeping track of `state`. Return `true` if
-traversal finished normally; `false` if one of the visit functions returned `false` (see )
+Traverse a graph `g` from source vertex/vertices `ss` keeping track of `state`. Return `true` if
+traversal finished normally; `false` if one of the visit functions returned `false`.
 """
-traverse_graph!(g::AbstractGraph, s::Integer, alg, state) = traverse_graph!(g, [s], alg, state)
+function traverse_graph! end
 
-struct VisitState{T<:Integer} <: AbstractTraversalState
+struct VisitState{T<:Integer} <: TraversalState
     visited::Vector{T}
 end
 
 @inline function initfn!(s::VisitState, u)
     push!(s.visited, u)
     return true
 end
+
+# note: since `newvisitfn!(s, u, v, t)` defaults to calling `newvisitfn!(s, u, v)`, this function
+# by default creates the necessary visitor functions for parallel traversals.
 @inline function newvisitfn!(s::VisitState, u, v)
     push!(s.visited, v)
     return true
 end
 
 """
-    visited_vertices(g, s, alg)
     visited_vertices(g, ss, alg)
 
 Return a vector representing the vertices of `g` visited in order by [`TraversalAlgorithm`](@ref) `alg`
-starting at vertex `s` (vertices `ss`).
+starting at vertex/vertices `ss`.
```

Comment:
which can be an integer or a collection of vertices?

Reply:
yes, I tried to get that across by using the singular `vertex` as well as the plural `vertices`. Is there a way to be more clear?

Reply:
As discussed on slack, we can make a docstring change very easily later if we need to.